feat: Add configuration option githubTokenWarn (#14902)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

- `ssh`: use SSH URLs provided by the platform for Git
- `endpoint`: ignore URLs provided by the platform and use the configured endpoint directly

## githubTokenWarn

> By default, Renovate will display a warning when `GITHUB_COM_TOKEN` is not set. By configuring this setting to `false`, Renovate will instead suppress these warnings on Pull Requests, etc. This is useful for self-hosted environments that don't have access to github.com and don't want to see this displayed when it's expected behavior to not have that configured.

Reviewer: ```suggestion
By default, Renovate will log and display a warning when `GITHUB_COM_TOKEN` is not set.
By configuring this setting to `false`, Renovate will instead suppress these warnings on Pull Requests, etc.
This is useful for self-hosted environments that don't have access to github.com and don't want to see this displayed when it's expected behavior to not have that configured.
```

- `ssh`: use SSH URLs provided by the platform for Git
- `endpoint`: ignore URLs provided by the platform and use the configured endpoint directly

## githubTokenWarn

> By default, Renovate will log and display a warning when `GITHUB_COM_TOKEN` is not set.

Reviewer 2: ```suggestion
By default, Renovate logs and displays a warning when the `GITHUB_COM_TOKEN` is not set.
```

> By configuring this setting to `false`, Renovate will instead suppress these warnings on Pull Requests, etc.

Reviewer 2: ```suggestion
By setting `githubTokenWarn` to `false`, Renovate suppresses these warnings on Pull Requests, etc.
```

> This is useful for self-hosted environments that don't have access to github.com and don't want to see this displayed when it's expected behavior to not have that configured.

Reviewer 2: ```suggestion
Disabling the warning is helpful for self-hosted environments that can't access the `github.com` domain, because the warning is useless in these environments.
```

How about something like this instead? The original sentence is hard to understand.
